newsroom/auth_server/auth.py

```diff
@@ -0,0 +1,32 @@
+from time import time
+from authlib.jose import jwt
+from authlib.jose.errors import BadSignatureError, ExpiredTokenError, DecodeError
+from flask import current_app as app
+from eve.auth import TokenAuth
+
+
+class JWTAuth(TokenAuth):
+    """
+    Implements JWT auth logic.
+    """
+
+    def check_auth(self, token, allowed_roles, resource, method):
+        """
+        This function is called to check if a token is valid. Must be
+        overridden with custom logic.
+        :param token: token.
+        :param allowed_roles: allowed user roles.
+        :param resource: resource being requested.
+        :param method: HTTP method being executed (POST, GET, etc.)
+        """
+        if not app.config.get("AUTH_SERVER_SHARED_SECRET"):
```

Comment:
I would probably log a warning here that it's not configured and thus clients can't use it